core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -30,6 +30,15 @@ private CatalogProperties() {
   public static final String FILE_IO_IMPL = "io-impl";
   public static final String WAREHOUSE_LOCATION = "warehouse";
 
+  public static final String TABLE_CACHE_ENABLED = "cache-enabled";
```

Comment:
why `TABLE_CACHE_` instead of just `CACHE_`?

Reply:
I originally thought the interval will work in combination with `cache-enabled`, but now we are can fully determine if caching is determined based on the expiration interval, so this is really just a shortcut. This is definitely a much cleaner solution.

This reminds me of how we decided to make `type` is just a shortcut of `catalog-impl`, and I think we should just follow what we did there.

So We don't need  `cache-enabled` as a catalog property, and instead it just translates to an interval value at engine side to initialize the catalog. I think this will simplify your Spark tests a lot, as there is no need for you to test all those combinations of `cache-enabled` and interval settings. You just need 1 test for the translation, and ensure the interval works correctly at core level.

Reply:
Also, `TABLE_CACHE` had not much basis in anything other than to distinguish from other potential caches. I'm happy to drop the TABLE prefix and just use `CACHE` if we'd like.

Reply:
Well, I'm leaving the `cache-enabled` in one of my tests, as the `cache.expiration-interval-ms` of 0 will override the value of `cache-enabled`.

So I think we should keep it, as well for backwards compatibility. But it did simplify my tests a bit =)

Reply:
I did however drop the `TABLE_` prefix =)

Reply:
thanks, I think if we want to have sub-level caching we can always add those like `cache.table.xxx`, `cache.partition.xxx`, but this can always be the top level default config.

Reply:
That makes sense to me. For now this is the only caching that happens so the extra prefix is unnecessary.